modules/accounts/calendar.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.accounts.calendar;
+
+  calendarOpts = { name, config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        readOnly = true;
+        description = ''
+          Unique identifier of the calendar. This is set to the
+          attribute name of the calendar configuration.
+        '';
+      };
+
+      path = mkOption {
+        type = types.str;
+        default = "${cfg.basePath}/${name}";
+        description = "The path of the storage.";
+      };
+
+      primary = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether this is the primary account. Only one account may be
+          set as primary.
+        '';
+      };
+
+      primaryCollection = mkOption {
+        type = types.str;
+        default = if (config.vdirsyncer.collections == null || config.vdirsyncer.collections == []) then
```

Comment:
Ideally defaults in this general module should not be defined in terms of values people set in more specific modules. I would suggest having no default.